### PR TITLE
Fixes #28294 - Create repo with ssl cert name

### PR DESCRIPTION
**Pull request by johnpmitsch**
This adds the ability to create a repository with a content credential names, not just ids

For example:

hammer repository create --product-id 3 --name my_repo --url https://example.com/myawesomerepo --content-type yum --ssl-client-key-name "my product key" --ssl-client-cert-name "my product cert" --ssl-ca-cert-name "my ca"

### CODE REVIEW

**Comment by johnpmitsch** [comment]
To test:

1. Create 3 content credentials, these can be actual credentials, but for the purpose of testing out parameters (and not actually syncing w/ credentials) you can use a random string to fake out the content credential
2. checkout PR and run `FOREMAN_APIPIE_LANGS=en bundle exec rake apipie:cache` on the katello box and then run `hammer -r` on a hammer devel box or wherever you have hammer set up.
3. Create a repository with the content credential names passed in, for example: `hammer repository create --product-id 3 --name my_repo --url https://example.com/myawesomerepo --content-type yum --ssl-client-key-name "my product key" --ssl-client-cert-name "my product cert" --ssl-ca-cert-name "my ca"`

Bad input is also handled:
```
[vagrant@centos7-hammer-devel ~]$ hammer repository create --product-id 3 --name ssl_by_name8 --content-type yum --ssl-client-key-name "rh product keyasdf" --ssl-client-cert-name "rh product certasdf"
Could not create the repository:
  Couldn't find ssl_client_cert with name 'rh product certasdf'

[vagrant@centos7-hammer-devel ~]$ hammer repository create --product-id 3 --name ssl_by_name8 --content-type yum --ssl-client-key-name "rh product keyasdf" --ssl-client-cert-name "rh product cert"
Could not create the repository:
  Couldn't find ssl_client_key with name 'rh product keyasdf'

[vagrant@centos7-hammer-devel ~]$ hammer repository create --product-id 3 --name RH_RHEL8 --url "" --content-type yum --ssl-client-key-name "rh product key" --ssl-client-cert-name "rh product cert" --ssl-ca-cert-name "rh caasdfa"
Could not create the repository:
  Couldn't find ssl_ca_cert with the name 'rh caasdfa'
```

File: app/controllers/katello/api/v2/repositories_controller.rb [hunk 1]
```diff
@@ -35,9 +35,12 @@ def custom_index_relation(collection)
...
       param :gpg_key_id, :number, :desc => N_("id of the gpg key that will be assigned to the new repository")
-      param :ssl_ca_cert_id, :number, :desc => N_("Idenifier of the SSL CA Cert")
-      param :ssl_client_cert_id, :number, :desc => N_("Identifier of the SSL Client Cert")
-      param :ssl_client_key_id, :number, :desc => N_("Identifier of the SSL Client Key")
+      param :ssl_ca_cert_id, :number, :desc => N_("Identifier of the SSL CA Cert content credential")
+      param :ssl_client_cert_id, :number, :desc => N_("Identifier of the SSL Client Cert content credential")
+      param :ssl_client_key_id, :number, :desc => N_("Identifier of the SSL Client Key content credential")
+      param :ssl_ca_cert_name, String, :desc => N_("Name of the SSL CA Cert content credential")
+      param :ssl_client_cert_name, String, :desc => N_("Name of the SSL Client Cert content credential")
+      param :ssl_client_key_name, String, :desc => N_("Name of the SSL Client Key content credential")
```
Comment (jlsherrill) [inline app/controllers/katello/api/v2/repositories_controller.rb]: hrmmm, we normally solve this in hammer via a name resolver.  Any reason to not do that here?

Comment (johnpmitsch) [reply app/controllers/katello/api/v2/repositories_controller.rb]: Thats probably a better way to go about this, I totally forgot about that aspect of hammer. :see_no_evil: I'll open up a PR there!

**Comment by johnpmitsch** [comment]
closing in favor of https://github.com/Katello/hammer-cli-katello/pull/720

**Pull request closed without merging**